lib/spack/spack/config.py

```diff
@@ -774,13 +854,98 @@ def override(
 
 def _add_platform_scope(cfg: Configuration, name: str, path: str, writable: bool = True) -> None:
     """Add a platform-specific subdirectory for the current platform."""
+    import spack.platforms  # circular dependency
+
     platform = spack.platforms.host().name
     scope = DirectoryConfigScope(
         f"{name}/{platform}", os.path.join(path, platform), writable=writable
     )
     cfg.push_scope(scope)
 
 
+#: Data class for the relevance of an optional path conditioned on either
+#: a spec and or explicitly specified as optional.
+class OptionalPath(NamedTuple):
+    path: str
+    when: str
+    optional: bool
+
+
+def included_path(entry: Union[str, dict]) -> OptionalPath:
+    """Convert the included path entry into a standard form.
```

Comment:
I think "standard form" means an `OptionalPath` (or rather `IncludePath`, but I would try not to use words like "standard" b/c it's not clear exactly what it means (unless there's actually a standard somewhere, in which case I'd expect a link).

```suggestion
    """Convert the included path entry into an IncludePath.
```